packages/contracts/test/voting/multisig.ts

```diff
@@ -338,140 +434,189 @@ describe('Multisig', function () {
-      {
-        const event = await findEvent(tx, DAO_EVENTS.EXECUTED);
+        const event = await findEvent(tx, 'Approved');
+        expect(event.args.proposalId).to.eq(id);
+        expect(event.args.approver).to.not.eq(multisig.address);
+        expect(event.args.approver).to.eq(signers[0].address);
 
-        expect(event.args.actor).to.equal(multisig.address);
-        expect(event.args.callId).to.equal(id);
-        expect(event.args.actions.length).to.equal(1);
-        expect(event.args.actions[0].to).to.equal(dummyActions[0].to);
-        expect(event.args.actions[0].value).to.equal(dummyActions[0].value);
-        expect(event.args.actions[0].data).to.equal(dummyActions[0].data);
-        expect(event.args.execResults).to.deep.equal(['0x']);
+        expect((await multisig.getProposal(id)).tally.approvals).to.equal(1);
+      });
+    });
 
-        const prop = await multisig.getProposal(id);
-        expect(prop.executed).to.equal(true);
-      }
+    describe('canExecute:', async () => {
+      it('returns `false` if the proposal is already executed', async () => {
+        await multisig.connect(signers[0]).approve(id, false);
+        await multisig.connect(signers[1]).approve(id, false);
+        await multisig.connect(signers[2]).approve(id, true);
 
-      // check for the `ProposalExecuted` event in the voting contract
-      {
-        const event = await findEvent(tx, VOTING_EVENTS.PROPOSAL_EXECUTED);
-        expect(event.args.proposalId).to.equal(id);
-      }
+        expect((await multisig.getProposal(id)).executed).to.be.true;
 
-      // calling execute again should fail
-      await expect(multisig.execute(id)).to.be.revertedWith(
-        customError('ProposalExecutionForbidden', id)
-      );
-    });
-  });
+        expect(await multisig.canExecute(id)).to.be.false;
+      });
+
+      it('returns `true` if the proposal can be executed', async () => {
+        await multisig.connect(signers[0]).approve(id, false);
+        await multisig.connect(signers[1]).approve(id, false);
+        await multisig.connect(signers[2]).approve(id, false);
 
-  describe('MultisigSettings', async () => {
-    it('should set the right minApprovals during initialization', async () => {
-      await multisig.initialize(dao.address, addresslist(5), {
-        onlyListed,
-        minApprovals,
+        expect((await multisig.getProposal(id)).executed).to.be.false;
+
+        expect(await multisig.canExecute(id)).to.be.true;
       });
-      expect((await multisig.multisigSettings()).minApprovals).to.be.eq(
-        minApprovals
-      );
     });
 
-    it('should set the right onlyListed during initialization', async () => {
-      await multisig.initialize(dao.address, addresslist(5), {
-        onlyListed,
-        minApprovals,
+    describe('execute:', async () => {
+      it('executes if the minimum approval is met', async () => {
+        await multisig.connect(signers[0]).approve(id, false);
```

Comment:
Shouldn't there be a check for execute before all 3 approvals, and assert that it reverts then?

Reply:
Added a test to check this separately f6ef2ab2e12930b80758c64f0ac107b6c1248a0d